### PR TITLE
Add model_override() method to Flow class for easy LLM model switching

### PR DESCRIPTION
**Pull request by shivchander**
## Summary
This PR adds a new `model_override()` method to the Flow class that allows users to easily override model configurations for LLM blocks in a flow, addressing the need for easy model switching without modifying YAML files.

**Key features:**
- Auto-detects LLM blocks by checking for model-related attributes (model, api_base, api_key)
- Supports overriding all detected LLM blocks or targeting specific blocks
- In-place modification with comprehensive parameter support
- Robust error handling and comprehensive logging

## Changes Made
- ✅ Added `model_override()` method to Flow class (`src/sdg_hub/core/flow/base.py`)
- ✅ Added `_detect_llm_blocks()` helper method for attribute-based LLM block detection
- ✅ Added comprehensive test coverage with 14 new tests (`tests/flow/test_base.py`)
- ✅ Improved code formatting consistency

## Usage Example
```python
from sdg_hub.core.flow import FlowRegistry, Flow

# Load your flow
flow_name = "Advanced Document Grounded Question-Answer Generation Flow for Knowledge Tuning"
flow_path = FlowRegistry.get_flow_path(flow_name)
flow = Flow.from_yaml(flow_path)

# Override model for all LLM blocks
flow.model_override(
    model="hosted_vllm/openai/gpt-oss-120b",
    api_base="http://localhost:8101/v1",
    api_key="EMPTY"
)

# Run with your new model
result = flow.generate(dataset)

# Reset by reloading from YAML if needed
flow = Flow.from_yaml(flow_path)
```

## Test Coverage
Added 14 comprehensive tests covering:
- ✅ LLM block detection by different attributes (model, api_base, api_key)
- ✅ Auto-detection vs. specific block targeting
- ✅ Partial parameter overrides
- ✅ Error cases (no parameters, invalid block names)
- ✅ Edge cases (no LLM blocks, missing attributes)
- ✅ Preservation of unchanged attributes

## Test Results
```bash
$ pytest tests/flow/test_base.py -k "test_detect_llm_blocks or test_model_override" -v
========================= 14 passed, 30 deselected, 1 warning in 0.02s =========================

$ pytest tests/flow/test_base.py -v
========================= 44 passed, 1 warning in 0.07s =========================
```

## Breaking Changes
None - this is a pure addition to the existing API.

## Documentation
The method includes comprehensive docstrings with examples and parameter descriptions.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/sdg_hub/core/flow/base.py [hunk 1]
```diff
@@ -427,6 +441,173 @@ def _prepare_block_kwargs(
         """Prepare execution parameters for a block."""
         return runtime_params.get(block.block_name, {})
 
+    def model_override(
```
Comment (eshwarprasadS) [inline src/sdg_hub/core/flow/base.py]: Thanks for adding this @shivchander. One thought here is that, this PR introduces a pattern which may very well become the default / predominant way to run generation, since the alternative is to mention client attributes per block in the yaml, which I assume many users might want to avoid.

Given this, should we consider a more default-sounding API name? along the lines of `configure_model()` or `set_model_config()`, or `set_client()`?

Comment (shivchander) [reply src/sdg_hub/core/flow/base.py]: Good point, I have updated the PR with to reflect this better

**Review by eshwarprasadS — APPROVED** [review]
Looks great thanks
